src/main/java/org/breedinginsight/api/v1/controller/OntologyController.java

```diff
@@ -100,23 +96,25 @@ public HttpResponse<Response<DataResponse<SharedProgram>>> getAvailablePrograms(
     @Post("/programs/{programId}/ontology/shared/programs")
     @Produces(MediaType.APPLICATION_JSON)
     @ProgramSecured(roles = {ProgramSecuredRole.BREEDER})
-    public HttpResponse<Response<DataResponse<SharedProgram>>> shareOntology(
+    public HttpResponse<Response<DataResponse<SharedOntology>>> shareOntology(
             @PathVariable UUID programId, @Body List<SharedOntologyProgramRequest> request) {
         try {
-            List<SharedProgram> sharedPrograms = ontologyService.shareOntology(programId, securityService.getUser(), request);
-            List<org.breedinginsight.api.model.v1.response.metadata.Status> metadataStatus = new ArrayList<>();
+            List<SharedOntology> sharedOntologies = ontologyService.shareOntology(programId, securityService.getUser(), request);
+            List<Status> metadataStatus = new ArrayList<>();
             metadataStatus.add(new Status(StatusCode.INFO, "Successful Creation"));
-            Pagination pagination = new Pagination(sharedPrograms.size(), 1, 1, 0);
+            Pagination pagination = new Pagination(sharedOntologies.size(), 1, 1, 0);
             Metadata metadata = new Metadata(pagination, metadataStatus);
-            Response<DataResponse<SharedProgram>> response = new Response(metadata, new DataResponse<>(sharedPrograms));
+            Response<DataResponse<SharedOntology>> response = new Response(metadata, new DataResponse<>(sharedOntologies));
             return HttpResponse.ok(response);
         } catch (ValidatorException e) {
             log.error("Validation errors", e);
             HttpResponse response = HttpResponse.status(HttpStatus.UNPROCESSABLE_ENTITY).body(e.getErrors());
             return response;
         } catch (UnprocessableEntityException e) {
+            log.error(e.getMessage());
```

Comment:
Sorry to be a pain, but for these error messages, can you change this to `log.error(e.getMessage(), e);`, please?  That way the error stack gets printed to the logs too.

Reply:
Thanks for catching that! Added.